src/Core/src/Platform/iOS/TabbedViewExtensions.cs

```diff
@@ -118,5 +119,52 @@ internal static void UpdateiOS15TabBarAppearance(
 			// Set the TabBarAppearance
 			tabBar.StandardAppearance = tabBar.ScrollEdgeAppearance = _tabBarAppearance;
 		}
+
+		internal static UIImage? AutoResizeTabBarImage(UITraitCollection traitCollection, UIImage image)
+		{
+			if (image == null || image.Size.Width <= 0 || image.Size.Height <= 0)
+			{
+				return null;
+			}
+
+			CGSize newSize = image.Size;
+
+			//Define size constants according to Apple's guidelines
+			//https://developer.apple.com/design/human-interface-guidelines/tab-bars#Target-dimensions
+			const int regularWideWidth = 31, compactWideWidth = 23;
+			const int regularTallHeight = 28, compactTallHeight = 20;
+			const int regularSquareSize = 25, compactSquareSize = 18;
+
+			bool isRegularTabBar = traitCollection.VerticalSizeClass == UIUserInterfaceSizeClass.Regular;
+			if (image.Size.Width > image.Size.Height) //Wide
+			{
+				newSize.Width = isRegularTabBar ? regularWideWidth : compactWideWidth;
+				newSize.Height = newSize.Width * image.Size.Height / image.Size.Width;
+			}
+			else if (image.Size.Width < image.Size.Height) //Tall
+			{
+				newSize.Height = isRegularTabBar ? regularTallHeight : compactTallHeight;
+				newSize.Width = newSize.Height * image.Size.Width / image.Size.Height;
+			}
+			else //Square
+			{
+				newSize.Width = isRegularTabBar ? regularSquareSize : compactSquareSize;
+				newSize.Height = newSize.Width;
+			}
+
+			UIImage? resizedImage = null;
+			try
+			{
+				UIGraphics.BeginImageContextWithOptions(newSize, false, 0);
+				image.Draw(new CGRect(0, 0, newSize.Width, newSize.Height));
```

Comment:
Can we follow the approach taken in this PR? 
https://github.com/dotnet/maui/pull/26016/files#diff-f405765594e76258ddbd4cb0d165c8ed0799223286a63b2d3a140bd02ccdbcb3R29-R31

Instead of doing Draw here we reuse the logic from button? 

We can either way for that PR to merge to 10.0

Or if you want to just copy those changes into this PR we can merge your PR and that other one will make it into net10.0 eventually